tls/README.md

```diff
@@ -0,0 +1,287 @@
+            },
+            "ClientCertDetails": null
+          }
+        },
+        "Status": {
+          "Errors": null
+        }
+      },
+```
+
+This stores the following info:
+* all secrets / configmaps which are certificates along with their metadata
+* parses them and stores metadata
+* removes locations that are versioned or hashed copies of existing certificates
+* deduplicates them by content, so copies of certificates are grouped by location
+
+## Certificate metadata
+
+Certificate contents may however be insufficient - i.e. it's not clear which OpenShift is responsible 
+for the certificate lifecycle. To do that the certs should have additional annotations.
+
+Recently we've added the `openshift.io/owning-component: foobar` annotation to (almost) all certificates
+so that issues related to this certificate would be routed to a proper location. For instance, 
+problems with `service-ca` certificates are filed in Jira for "Networking / cluster-network-operator" component 
+regardless of the repo these are created or the namespace its being created, as this component is 
+injecting CA bundle.
+
+Along with manually adding annotations `library-go` was updated to pass `JiraComponent` and `Description` 
+annotations for secrets when `RotatedSelfSignedCertKeySecret`, `RotatedSigningCASecret` or 
+`CABundleConfigMap` structs are used. These structs would be later updated to add more necessary 
+metadata - and users are encouraged to make use of them to keep up with metadata requirements.
+
+## Reports and violations
+
+After the TLS registry was generated by the e2e test, it can be analyzed to ensure it matches metadata requirements.
+The file in `rawTLSInfo` is copied to `tls/raw-data` in this repository so that it can be analyzed by 
+the `openshift-tests update-tls-artifacts` command.
+
+The command will parse the registry JSON and verify that certificates match a set of requirements.
+The most basic is "every certificate has to have an owner annotations". The processed registry JSON is 
+stored in `tls/ownership/ownership.json` so that it can be machine-readable.
+
+Along with JSON human-readable report in Markdown at `tls/ownership/ownership.md` is generated. 
+This report contains a list of certs violating this requirement and a list of certs 
+grouped by owning components along with their description. 
+These kinds of reports are useful to find the owning component for specific cert.
+
+Apart from reports the registry is checked for requirement violations. 
+`make update` creates `tls/violations/ownership/ownership-violations.json` listing certificate locations
+  that don't have ownership annotation set. This file is meant to be "remove-only", meaning adding 
+  new entries is prohibited. This is enforced by using a separate `OWNERS` file for this directory and an e2e test (see below).
+
+## Adding a new requirement
+
+Reports and violations mechanisms can be extended to add new requirements. To add a new 
```

Comment:
Provide our next examples. Description and auto-regenerate after offline expiry.